fe/fe-core/src/main/java/org/apache/doris/catalog/Column.java

```diff
@@ -386,6 +386,14 @@ public void checkSchemaChangeAllowed(Column other) throws DdlException {
             throw new DdlException("Can not change " + getDataType() + " to " + other.getDataType());
         }
 
+        if (type.isNumericType() && other.type.isStringType()) {
+            Integer lSize = type.getColumnStringRepSize();
+            Integer rSize = other.type.getColumnStringRepSize();
+            if (rSize < lSize) {
+                throw new DdlException("Can not change from wider type to narrower type");
```

Comment:
Add more detail info in error msg, such as column name, lSize, rSize.
So that user can easy debugging.